cuda/delete.cpp

```diff
@@ -0,0 +1,95 @@
+#include <bits/stdc++.h>
+
+#include "kernels.h"
+
+using namespace std;
+
+void delete_point(Point point_to_delete, Grid *root_grid,
+				  vector<QuadrantBoundary> &boundaries,
+				  unordered_map<int, Grid *> &grid_map, int quadrant_id) {
+	// Access the target grid from the unordered map
+	Grid *target_grid = grid_map[quadrant_id];
```

Comment:
maybe we can just pass target_grid rather than calling grid_map again

Reply:
Commented on this previously.